File: python/misc/factorial.py
import sys

def fact(i):
  def fact2(i, acc):
    if i:
      return fact2(i-1, i*acc)
    else:
      return acc

  return fact2(i, 1)

if __name__ == "__main__":
  i = int(sys.argv[1])
  print fact(i)

File: erlang/factorial.erl
%#!/usr/bin/env escript
-mode(compile).

main([X]) ->
  io:format("~p~n", [fact(list_to_integer(X))]).

fact(I) ->
  fact2(I, 1).

fact2(0, Acc) ->
  Acc;
fact2(I, Acc) ->
  fact2(I-1, I*Acc).
